### PR TITLE
Fix dynamic array assignment to unpacked arrays (#6907)

### LINKED ISSUE #6907: Error when copying dynamic array into fixed-size array of same size
In regressions of cases from the SystemVerilog LRM, legal case on 1800-2012 page 116-117 (or 1800-2023 page 161) errors out:

%Error: VARREF 'sv12_lrm_p0117_int_c_new.B' is not an unpacked array, but is in an unpacked array context
%Error: ARRAYSEL is not an unpacked array, but is in an unpacked array context

**Code:**
```
int A[2][100:1];
int B[] = new[100];
int D [3][][];

initial begin
    A[1] = B;
    A = D[2];
end
```

**Version:**
5.044

**Environment:**
Centos 7.9，GCC 11.5

### PR DESCRIPTION
**Pull request by skku970412**
Fixes #6907.

This keeps fixed unpacked array assignments from trying to slice dynamic array and queue RHS values, and emits the existing array conversion path backed by VlQueue-to-VlUnpacked copy helpers.

Tests:
- VERILATOR_ROOT=$PWD test_regress/t/t_dynarray.py
- VERILATOR_ROOT=$PWD test_regress/t/t_queue_assignment.py
- VERILATOR_ROOT=$PWD test_regress/t/t_unpacked_to_queue.py
- VERILATOR_ROOT=$PWD test_regress/t/t_stream_queue.py
- VERILATOR_ROOT=$PWD test_regress/t/t_stream_dynamic.py
- make -j20
- make format
- make cppcheck
- make lint-py

### CODE REVIEW

File: test_regress/t/t_dynarray.v [hunk 1]
```diff
@@ -26,6 +26,8 @@ module t (
   typedef bit [7:0] byte_t;
   byte_t a[];
   byte_t b[];
+  byte_t fixed[256];
```
Comment (wsnyder) [inline test_regress/t/t_dynarray.v:29]: Need to also test reverse-endian and non-zero lsb e.g. `fixed[1:256]`.

File: include/verilated_funcs.h [hunk 2]
```diff
@@ -3086,6 +3086,27 @@ static inline void VL_REVCOPY_Q(VlQueue<T>& q, const VlUnpacked<T, N_Depth>& fro
...
+// Overloads for VlQueue source -> VlUnpacked destination
+template <typename T, std::size_t N_Depth>
+static inline void VL_COPY_Q(VlUnpacked<T, N_Depth>& q, const VlQueue<T>& from, int lbits,
+                             int srcElementBits, int dstElementBits) {
+    if (srcElementBits == dstElementBits) {
+        for (size_t i = 0; i < N_Depth; ++i) q[i] = from.at(static_cast<int32_t>(i));
+    } else {
+        VlQueue<T> dstQ;
+        VL_COPY_Q(dstQ, from, lbits, srcElementBits, dstElementBits);
+        for (size_t i = 0; i < N_Depth; ++i) q[i] = dstQ.at(static_cast<int32_t>(i));
```
Comment (wsnyder) [inline include/verilated_funcs.h]:
```suggestion
        VlQueue<T> tmpQ;
        VL_COPY_Q(tmpQ, from, lbits, srcElementBits, dstElementBits);
        for (size_t i = 0; i < N_Depth; ++i) q[i] = tmpQ.at(static_cast<int32_t>(i));
```

File: src/V3Width.cpp [hunk 3]
```diff
@@ -6561,6 +6562,30 @@ class WidthVisitor final : public VNVisitor {
...
+        AstNodeExpr* const rhsp = nodep->rhsp();
+        AstNodeDType* const rhsDTypep = rhsp->dtypep()->skipRefp();
+        if (!VN_IS(rhsDTypep, DynArrayDType) && !VN_IS(rhsDTypep, QueueDType)) return;
+
+        int srcElementBits = 1;
+        if (const AstNodeDType* const elemDtp = rhsDTypep->subDTypep()) {
+            srcElementBits = elemDtp->width();
+        }
+        int dstElementBits = 1;
+        if (const AstNodeDType* const elemDtp = lhsDTypep->subDTypep()) {
```
Comment (wsnyder) [inline src/V3Width.cpp]:
```suggestion
        if (const AstNodeDType* const elemDtp = rhsDTypep->subDTypep()->skipRefp()) {
            srcElementBits = elemDtp->width();
        }
        int dstElementBits = 1;
        if (const AstNodeDType* const elemDtp = lhsDTypep->subDTypep()->skipRefp()) {
```

**Comment by wsnyder** [comment]
Looks like this got stale, if you're still interested in this can you fix the conflicts please, and we'll work towards getting it in.

**Comment by wsnyder** [comment]
Closing as stale, please consider submitting a new PR to fix this.
